### PR TITLE
2293 add page for upgrade tags

### LINKED ISSUE #2293: [BC Idea]: Add page for upgrade tags
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=6cced5dc-4698-ef11-95f5-6045bdb6c779

### Description

It would be very welcome to add a page to the upgrade tags table so in that case you can see what kind of upgrade tags there are executed for troubleshooting.

Refreclting to the discussion with @nikolakukrika on Yammer
https://www.yammer.com/dynamicsnavdev/threads/3043376554098688

### I will provide the implementation for this BC Idea

- [x] I will provide the implementation for this BC Idea

### PR DESCRIPTION
**Pull request by Bertverbeek4PS**
#### Summary
It would be very welcome to add a page to the upgrade tags table so in that case you can see what kind of upgrade tags there are executed for troubleshooting.

#### Work Item(s) 
Fixes #2293

Fixes [AB#556569](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/556569)

### CODE REVIEW

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    DeleteAllowed = false;
+    ModifyAllowed = false;
+
+    layout
+    {
+        area(Content)
+        {
+            repeater(GroupName)
+            {
+                field(Tag; rec.Tag) { }
```
Comment (pri-kise) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al]: Could you please fix the casing to `Rec.`

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Upgrade;
+page 9998 PageName
```
Comment (pri-kise) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al]: PageName

**Review by pri-kise — CHANGES_REQUESTED** [review]

**Comment by JesperSchulz** [comment]
@nikolakukrika, could you give this a review?

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    layout
+    {
+        area(Content)
+        {
+            repeater(GroupName)
+            {
+                field(Tag; Rec.Tag) { }
+                field("Tag Timestamp"; Rec."Tag Timestamp") { }
+                field(Company; Rec.Company) { }
```
Comment (nikolakukrika) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al:29]: We can add more fields, like skipped and the date it was created/modified. They can also be hidden and personalized onto the page.

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Upgrade;
+page 9998 "Upgrade Tags"
+{
+    PageType = List;
+    ApplicationArea = All;
+    Caption = 'Upgrade Tags';
```
Comment (nikolakukrika) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al:11]: We are missing extensible false

**Review by nikolakukrika — CHANGES_REQUESTED** [review]

**Comment by nikolakukrika** [comment]
Awesome, thanks for doing this, please just add the Extensible false to the page.

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al (hunk 3 above)
Comment (Bertverbeek4PS) [reply src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al:29]: Added also the skipped field.
the system audit fields they can put it on by personalisation

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 5]
```diff
@@ -0,0 +1,32 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Upgrade;
+page 9998 "Upgrade Tags"
```
Comment (JesperSchulz) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al]: All other objects in this app have inherent permissions/entitlements. I assume this page needs that too?

**Review by JesperSchulz — CHANGES_REQUESTED** [review]

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+{
+    PageType = List;
+    ApplicationArea = All;
+    Caption = 'Upgrade Tags';
+    Extensible = false;
+    UsageCategory = Lists;
+    SourceTable = "Upgrade Tags";
+    InsertAllowed = false;
+    DeleteAllowed = false;
+    ModifyAllowed = false;
```
Comment (JesperSchulz) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al:17]:
```suggestion
    ModifyAllowed = false;
    InherentEntitlements = X;
    InherentPermissions = X;
```

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al (hunk 5 above)
Comment (JesperSchulz) [reply src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al]: I will deal with that :-)

**Review by JesperSchulz — APPROVED** [review]

**Review by nikolakukrika — DISMISSED** [review]

File: src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al [hunk 7]
```diff
@@ -0,0 +1,34 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Upgrade;
+page 9998 "Upgrade Tags"
```
Comment (JesperSchulz) [inline src/System Application/App/Upgrade Tags/src/UpgradeTags.Page.al]:
```suggestion
page 9985 "Upgrade Tags"
```

**Review by JesperSchulz — APPROVED** [review]

**Review by nikolakukrika — APPROVED** [review]

**Comment by Bertverbeek4PS** [comment]
thanks @JesperSchulz and @nikolakukrika !!

**Comment by JesperSchulz** [comment]
> thanks @JesperSchulz and @nikolakukrika !!

Thank YOU for this great contribution 🥳

**Pull request merged**